pypfopt/base_optimizer.py

```diff
@@ -218,8 +221,32 @@ def _map_bounds_to_constraints(self, test_bounds):
                 self._lower_bounds = np.nan_to_num(lower, nan=-1)
                 self._upper_bounds = np.nan_to_num(upper, nan=1)
 
-        self._constraints.append(self._w >= self._lower_bounds)
-        self._constraints.append(self._w <= self._upper_bounds)
+        self.add_constraint(lambda x: x >= self._lower_bounds)
+        self.add_constraint(lambda x: x <= self._upper_bounds)
+
+    def is_parameter_defined(self, parameter_name: str) -> bool:
+        is_defined = False
+        objective_and_constraints = self._constraints + [self._objective] if self._objective is not None else self._constraints
+        for expr in objective_and_constraints:
+            params = [arg for arg in get_all_args(expr) if isinstance(arg, cp.Parameter)]
+            for param in params:
+                if param.name() == parameter_name and not is_defined:
+                    is_defined = True
+                elif param.name() == parameter_name and is_defined:
+                    raise Exception('Parameter name defined multiple times')
+        return is_defined
+
+    def update_parameter_value(self, parameter_name: str, new_value: float) -> None:
+        assert self.is_parameter_defined(parameter_name)
+        was_updated = False
+        objective_and_constraints = self._constraints + [self._objective] if self._objective is not None else self._constraints
```

Comment:
For quadratic utility, the parameter is in the objective, thus we need to check both. The functions `get_all_args()` and `flatten()` could probably be moved somewhere else.